server/services/visitSessionsService.ts

```diff
@@ -205,14 +209,46 @@ export default class VisitSessionsService {
     const token = await this.systemToken(username)
     const visitSchedulerApiClient = this.visitSchedulerApiClientBuilder(token)
 
-    const visits = await visitSchedulerApiClient.getUpcomingVisits(offenderNo)
     logger.info(`Get upcoming visits for ${offenderNo}`)
+    const visits = await visitSchedulerApiClient.getUpcomingVisits(offenderNo)
 
     return visits.map(visit => {
       return this.buildVisitInformation(visit)
     })
   }
 
+  async getFullVisitDetails({
+    username,
+    reference,
+  }: {
+    username: string
+    reference: string
+  }): Promise<{ visit: Visit; visitors: Contact[] }> {
+    const token = await this.systemToken(username)
+    const visitSchedulerApiClient = this.visitSchedulerApiClientBuilder(token)
+    const prisonerContactRegistryApiClient = this.prisonerContactRegistryApiClientBuilder(token)
+
+    logger.info(`Get visit ${reference}`)
+    const visit = await visitSchedulerApiClient.getVisit(reference)
+
+    logger.info(`Get contacts for ${visit.prisonerId}`)
+    const contacts = await prisonerContactRegistryApiClient.getPrisonerSocialContacts(visit.prisonerId)
+
+    const visitorIds: number[] = visit.visitors.reduce((acc, visitor) => {
+      acc.push(visitor.nomisPersonId)
+      return acc
+    }, [])
+
+    const visitors: Contact[] = contacts.reduce((acc, contact) => {
+      if (visitorIds.includes(contact.personId)) {
```

Comment:
Could possibly to a `.filter()` on this rather than the `if` statement?

Reply:
Good suggestion, will give that a try 👍